src/app/models/submission.server.model.ts

```diff
@@ -304,18 +305,21 @@ const getSubmissionCursorByFormId: IEncryptSubmissionModel['getSubmissionCursorB
     form: formId,
     ...createQueryWithDateParam(dateRange?.startDate, dateRange?.endDate),
   }
-  return this.find(streamQuery)
-    .select({
-      encryptedContent: 1,
-      verifiedContent: 1,
-      attachmentMetadata: 1,
-      created: 1,
-      id: 1,
-    })
-    .batchSize(2000)
-    .read('secondary')
-    .lean()
-    .cursor()
+  return (
+    this.find(streamQuery)
+      .select({
+        encryptedContent: 1,
+        verifiedContent: 1,
+        attachmentMetadata: 1,
+        created: 1,
+        id: 1,
+      })
+      .batchSize(2000)
+      .read('secondary')
+      .lean()
+      // Override typing as Map is converted to Object once passed through `lean()`.
+      .cursor() as QueryCursor<SubmissionCursorData>
```

Comment:
could we rename `SubmissionCursorData` to have the word `View`? for consistency with webhooks and public forms too

Reply:
also quick question - was `attachmentMetadata` already cast as an object before, or was this a change resulting from the upgrade?

Reply:
Previous typing was `any`, with the new stronger types from native mongoose we have to cast it

Reply:
The typing from mongoose was `QueryCursor<Partial<IFormSchema>>`, which had the typing of `Map` for `attachmentMetadata`. `SubmissionCursorData` has the type `Record`